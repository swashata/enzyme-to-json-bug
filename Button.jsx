import React from 'react';

const Button = props => (
	<button
		onClick={props.onClick}
		className={`button ${props.grouped ? 'is-grouped' : null}`}
	>
		Just a button
	</button>
);

export default Button;
