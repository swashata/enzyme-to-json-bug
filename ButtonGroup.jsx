import React from 'react';

const ButtonGroup = props => (
	<div className="button-group">
		{
			React.Children.map(props.children, child =>
				React.cloneElement(child, {
					grouped: true
				})
			)
		}
	</div>
);

export default ButtonGroup;
