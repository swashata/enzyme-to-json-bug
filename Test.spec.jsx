import React from 'react';
import {mount} from 'enzyme';

import Button from './Button';
import ButtonGroup from './ButtonGroup';

describe('This test will not pass', () => {
	const spy = jest.fn();
	const wrapper = mount(
		<ButtonGroup>
			<Button onClick={spy} />
			<Button onClick={spy} />
			<Button onClick={spy} />
			<Button onClick={spy} />
		</ButtonGroup>
	);

	test('does something on click', () => {
		// Comment this out, i.e do not simulate click
		// and this will pass
		wrapper.find('button').at(0).simulate('click');
		expect(spy).toHaveBeenCalledTimes(1);
	});

	test('matches snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	});
});
