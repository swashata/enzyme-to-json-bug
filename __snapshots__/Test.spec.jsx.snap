// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`This test will not pass matches snapshot 1`] = `
<ButtonGroup>
  <div
    className="button-group"
  >
    <Button
      grouped={true}
      key=".0"
      onClick={[MockFunction]}
    >
      <button
        className="button is-grouped"
        onClick={[MockFunction]}
      >
        Just a button
      </button>
    </Button>
    <Button
      grouped={true}
      key=".1"
      onClick={[MockFunction]}
    >
      <button
        className="button is-grouped"
        onClick={[MockFunction]}
      >
        Just a button
      </button>
    </Button>
    <Button
      grouped={true}
      key=".2"
      onClick={[MockFunction]}
    >
      <button
        className="button is-grouped"
        onClick={[MockFunction]}
      >
        Just a button
      </button>
    </Button>
    <Button
      grouped={true}
      key=".3"
      onClick={[MockFunction]}
    >
      <button
        className="button is-grouped"
        onClick={[MockFunction]}
      >
        Just a button
      </button>
    </Button>
  </div>
</ButtonGroup>
`;
